junifer/preprocess/warping/space_warper.py

```diff
@@ -0,0 +1,198 @@
+"""Provide class for warping data to other template spaces."""
+
+# Authors: Synchon Mandal <s.mandal@fz-juelich.de>
+# License: AGPL
+
+from typing import Any, ClassVar, Dict, List, Optional, Tuple, Type, Union
+
+from templateflow import api as tflow
+
+from ...api.decorators import register_preprocessor
+from ...utils import logger, raise_error
+from ..base import BasePreprocessor
+from ._ants_native_warper import ANTsNativeWarper
+from ._ants_template_warper import ANTsTemplateWarper
+from ._fsl_native_warper import FSLNativeWarper
+
+
+__all__ = ["SpaceWarper"]
+
+
+@register_preprocessor
+class SpaceWarper(BasePreprocessor):
+    """Class for warping data to other template spaces.
+
+    Parameters
+    ----------
+    using : {"fsl_native", "ants_native", "ants_template}
+        Implementation to use for warping:
+
+        * "fsl_native" : Use FSL's ``applywarp`` for native space warping
```

Comment:
Doesn't this need to happen automatically? I mean, why do the user need to change one parameter more than just the `reference`?

If `reference` is `"T1w"`, then we use the native one, if not, the template one.

I think this is one of the places where the user should not need to change more than the `reference` parameter.

Reply:
From our conversation above:
>> Gathering from your replies but quoting only this, I definitely support a single parameter; let's finalise it as method or using which goes in the constructor of every marker / preprocessor having this behaviour. I would want this parameter to be positional and not optional with no default value and no value as "auto". This would keep the metadata consistent. Now, after that the check happens as you already described above.
>
>100% agree.

I wouldn't want to make exception for any component. I agreed with you for `BOLDWarper` auto-detecting considering we had small scope. For this I disagree and my rationale is quoted.

> I think this is one of the places where the user should not need to change more than the reference parameter.

IMO it wouldn't hurt the user to add one extra line when in return it would give us complete knowledge of the component. I don't want to trade clarity for saving one line in the YAML.